src/System.Devices.Gpio/System.Devices.Gpio/UnixDriver.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Devices.Gpio
+{
+    public class UnixDriver : GpioDriver
+    {
+        private const string GpioPath = "/sys/class/gpio";
+
+        private ISet<int> _exportedPins;
+
+        public UnixDriver()
+        {
+            _exportedPins = new HashSet<int>();
+        }
+
+        public override void Dispose()
+        {
+            while (_exportedPins.Count > 0)
+            {
+                var pin = _exportedPins.First();
+                UnexportPin(pin);
+            }
+        }
+
+        public override GpioPinMode GetPinMode(int pin)
+        {
+            ExportPin(pin);
+
+            var directionPath = $"{GpioPath}/gpio{pin}/direction";
+            var stringMode = File.ReadAllText(directionPath);
```

Comment:
nit: the coding style that we use just uses `var` when the right side is something like `new T()` in this case and below we are explicit about the type (I know, I like `var` better, but consistency...)

Reply:
Ok, but for me is dificult to understand why C# designers added a feature (in this case var) if it is considered bad coding style haha. Do you know if there is some kind of Roslyn analyzer or similar that checks and warn (or even better, fix) my code to meet the coding style? That will be nice!